examples/geospatial-python-urban-analysis-with-postgis/src/ml/clustering.py

```
@@ -0,0 +1,67 @@
+import numpy as np
+import geopandas as gpd
+
+from sklearn.cluster import DBSCAN 
+from shapely.geometry import Point
+from rasterio.features import shapes
```

Ruff (0.8.2)

`rasterio.features.shapes` imported but unused

Remove unused import: `rasterio.features.shapes`

(F401)